### PR TITLE
Refactor fvm procedure output into a struct

### PR DESCRIPTION
**Pull request by pattyshack**
prep work to make fvm procedure idempotent

### CODE REVIEW

**Review by janezpodhostnik — APPROVED** [review]

File: fvm/fvm.go [hunk 1]
```diff
@@ -21,20 +23,62 @@ const (
 	ScriptProcedureType      = ProcedureType("script")
 )
 
+type ProcedureOutput struct {
+	// Output by both transaction and script.
+	Logs                   []string
+	Events                 flow.EventsList
+	ServiceEvents          flow.EventsList
+	ConvertedServiceEvents flow.ServiceEventList
```
[the remark this answers is not in the document]
Comment (pattyshack) [reply fvm/fvm.go:31]: script probably won't emit service events, but can emit events.  you've change the script env to support event output a few months back (in order to improve program reusability).

File: fvm/transactionInvoker.go [hunk 2]
```diff
@@ -454,26 +460,10 @@ func (executor *transactionExecutor) commit(
 	}
 
 	// if tx failed this will only contain fee deduction logs
```
[the remark this answers is not in the document]
Comment (pattyshack) [reply fvm/transactionInvoker.go:462]: thx

**Review by zhangchiqing — APPROVED** [review]

**Comment by pattyshack** [comment]
bors merge

File: fvm/script.go [hunk 3]
```diff
@@ -21,12 +20,9 @@ type ScriptProcedure struct {
 	Script         []byte
 	Arguments      [][]byte
 	RequestContext context.Context
-	Value          cadence.Value
-	Logs           []string
-	Events         []flow.Event
-	GasUsed        uint64
```
Comment (pattyshack) [inline fvm/script.go:27]: urgh ... need to keep this around for the emulator ...

**Comment by pattyshack** [comment]
bors merge

**Pull request merged**